prow/gerrit/adapter/adapter_test.go

```diff
@@ -970,6 +1023,63 @@ func TestProcessChange(t *testing.T) {
 			instance:     testInstance,
 			numPJ:        0,
 		},
+		{
+			name: "InRepoConfig Presubmits are retrieved",
+			change: client.ChangeInfo{
+				CurrentRevision: "1",
+				Project:         "test-infra",
+				Status:          "NEW",
+				Branch:          "inRepoConfig",
+				Revisions: map[string]client.RevisionInfo{
+					"1": {
+						Ref:     "refs/changes/00/1/1",
+						Created: stampNow,
+					},
+				},
+			},
+			instancesMap: map[string]*gerrit.AccountInfo{testInstance: {AccountID: 42}},
+			instance:     testInstance,
+			numPJ:        3,
+			pjRef:        "refs/changes/00/1/1",
+		},
+		{
+			name: "InRepoConfig Postsubmits are retrieved",
+			change: client.ChangeInfo{
+				CurrentRevision: "1",
+				Project:         "postsubmits-project",
+				Status:          "MERGED",
+				Branch:          "inRepoConfig",
+				Revisions: map[string]client.RevisionInfo{
+					"1": {
+						Ref:     "refs/changes/00/1/1",
+						Created: stampNow,
+					},
+				},
+			},
+			instancesMap: map[string]*gerrit.AccountInfo{testInstance: {AccountID: 42}},
+			instance:     testInstance,
+			numPJ:        2,
+			pjRef:        "refs/changes/00/1/1",
+		},
+		{
+			name: "no presubmit Prow jobs automatically triggered from WorkInProgess change. Works when clientFactory nil",
+			change: client.ChangeInfo{
+				CurrentRevision: "1",
+				Project:         "test-infra",
+				Status:          "NEW",
+				WorkInProgress:  true,
+				Revisions: map[string]gerrit.RevisionInfo{
+					"1": {
+						Number: 1001,
+					},
+				},
+			},
+			instancesMap:     map[string]*gerrit.AccountInfo{testInstance: {AccountID: 42}},
+			instance:         testInstance,
+			shouldError:      false,
+			numPJ:            0,
```

Comment:
there should also be a unit test fetching static jobs when gc is nil

Reply:
Done